### PR TITLE
BrowserGym example for LLMs (no vision)

### PR DESCRIPTION
**Pull request by sergiopaniego**
# What does this PR do?

This PR does GRPO on BrowserGym via the miniwob tasks. 

miniwob

TODO:

- [ ] Review system prompt so it works for both cases.
- [ ] Update BrowserGym env in OpenEnv if needed since we're currently using https://huggingface.co/spaces/burtenshaw/browsergym-v2

Launched like this:

```bash
python examples/scripts/openenv/browsergym.py --vllm-mode colocate --env-mode space --env-host https://burtenshaw-browsergym-v2.hf.space --model-id xxx/xxx --tokenizer-id xxx/xxx --gradient-accumulation-steps 1  --no-vlm
```
Reward curve for the this case:

Screenshot 2025-12-15 at 18 14 56

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

@burtenshaw @qgallouedec

### CODE REVIEW

**Comment by burtenshaw** [comment]
Nice work getting this running. This looks ok but I would make two improvements, if you have time:
- separate scripts for llm and vlm instead of parameter.
- remove the logic around openenv deployment and just handle urls w/o ports.

**Comment by sergiopaniego** [comment]
@burtenshaw super nice idea, updated!

File: examples/scripts/openenv/browsergym_llm.py [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+        return {
+            "prompt_ids": episode_prompt_ids,
+            "completion_ids": episode_completion_ids,
+            "logprobs": episode_logprobs,
+            "completion_reward": completion_rewards,
+        }
+
+    trainer = GRPOTrainer(
+        model=args.model_id,
+        processing_class=tokenizer,
```
Comment (qgallouedec) [inline examples/scripts/openenv/browsergym_llm.py]:
```suggestion
```

File: examples/scripts/openenv/browsergym_llm.py [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+# Main entrypoint
+# ---------------------------------------------------------------------------
+
+
+def main() -> None:
+    args = parse_args()
+
+    tokenizer = AutoTokenizer.from_pretrained(args.tokenizer_id)
+    tokenizer.pad_token = tokenizer.eos_token
+
```
Comment (qgallouedec) [inline examples/scripts/openenv/browsergym_llm.py]:
```suggestion
```

File: examples/scripts/openenv/browsergym_llm.py [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+from trl.experimental.openenv import generate_rollout_completions
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="Run GRPO training for BrowserGym MiniWoB using OpenEnv environment.")
+    parser.add_argument(
+        "--tokenizer-id",
+        default="Qwen/Qwen3-0.6B",
+        help="Model identifier used to load the tokenizer.",
+    )
```
Comment (qgallouedec) [inline examples/scripts/openenv/browsergym_llm.py]:
```suggestion
```
